xbmc/threads/platform/RecursiveMutex.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+#include <mutex>
+
+#if (defined TARGET_POSIX)
+#include <pthread.h>
+namespace XbmcThreads
+{
+  // forward declare in preparation for the friend declaration
+  class RecursiveMutex
+  {
+    pthread_mutex_t mutex;
```

Comment:
our coding guidelines require prefix "m_" for class members

Reply:
This entire class was restored from the PR #13454 that I deleted and you wanted reverted.
https://github.com/xbmc/xbmc/pull/13454/files#diff-d85cd2654abbc311f859cdb2fbb7ce4aL41